docs/enhancements/cluster_pool_hot_spares.md

```diff
@@ -0,0 +1,162 @@
+    - [User Stories](#user-stories)
+  - [Proposal](#proposal)
+    - [`spec.runningCount`](#specrunningcount)
+    - [Decouple PowerState on Creation](#decouple-powerstate-on-creation)
+    - [ClusterClaim Assignment by `creationTimestamp`](#clusterclaim-assignment-by-creationtimestamp)
+    - [Hibernation Controller Must Account For `hibernateAfter` in Pool Clusters](#hibernation-controller-must-account-for-hibernateafter-in-pool-clusters)
+  - [Alternatives](#alternatives)
+    - [Don't Hibernate New ClusterDeployments](#dont-hibernate-new-clusterdeployments)
+    - [`runningPercent`](#runningpercent)
+
+## Summary
+
+Support configuring ClusterPools to maintain some number of clusters (`0 <= N <= spec.size`) in Running state.
+
+## Motivation
+
+Today ClusterPools immediately hibernate new clusters.
+This means a ClusterClaim is guaranteed to suffer at least Resuming time before the cluster is available for use.
+While this is still faster than having to create a cluster from scratch, it is a potentially-unnecessary delay.
+As an example, CI runs could be reduced by ~5m if a ClusterClaim were able to grab a Running cluster rather than having to Resume a Hibernating one.
+
+### Goals
+
+Ability to tune a ClusterPool such that a ClusterClaim gets a Running cluster immediately in the common case.
+
+### Non-Goals
+
+Solving world peace.
+
+
+### User Stories
+
+1. As a consumer of a (properly tuned) ClusterPool,
+
+   I want ClusterClaim to give me a Running cluster immediately in the common case,
+
+   So that I can get my work done more quickly.
+
+   (Ideally without sacrificing the cost savings of having some clusters in the pool Hibernating.)
+
+2. As a user of a (hypothetical future) ClusterDeployment (or sibling CRD) for a cluster type that doesn't support hibernation,
+
+   I want to be able to configure a ClusterPool that never hibernates its ClusterDeployments
+
+   So that I can use ClusterPools for that kind of ClusterDeployment.
+
+## Proposal
+
+### `spec.runningCount`
+
+Add a field to ClusterPool indicating a portion of the pool to maintain in Running state.
+`spec.runningCount` is an integer absolute number of Running clusters to strive for.
+It is constrained to `0 <= runningCount <= Size`.
+We could enforce this by webhook; or we could just update the spec accordingly.
+
+If unspecified, the default behavior is to hibernate all CDs in the pool, as today.
+This corresponds to `runningCount=0`, conveniently the golang nil value of an int.
+
+### Decouple PowerState on Creation
+
+Today we create all new ClusterDeployments with `powerState=Hibernating` and update that to `Running` at claim time.
+With this feature, we will create with `powerState` unset.
+Then in a separate code path:
+- Sort unassigned (installing + ready) CDs by `creationTimestamp`
+- Update the oldest `runningCount` CDs with `powerState=Running`
+- Update the remainder (the newest `size - runningCount`) with `powerState=Hibernating`
```

Comment:
with this flow, some previously hibernated clusters will be made to transition to match the running count.
how do we handle case where the cluster fails to go from hibernated -> running ?

Reply:
Can we handle that generically in https://issues.redhat.com/browse/HIVE-1602

Reply:
> "ClusterPool should handle (trash and replace) broken CDs" -- which can really happen for any number of reasons, not just hibernate/resume.

One such reason being ProvisionStopped, per [HIVE-1615](https://issues.redhat.com/browse/HIVE-1615)/#1524.